react/features/base/tracks/middleware.js

```diff
@@ -159,18 +159,19 @@ function _getLocalTrack(store, mediaType: MEDIA_TYPE) {
  */
 function _setMuted(store, action, mediaType: MEDIA_TYPE) {
     const localTrack = _getLocalTrack(store, mediaType);
+    const muted = Boolean(action.muted);
 
     if (localTrack) {
-        setTrackMuted(localTrack.jitsiTrack, action.muted)
+        setTrackMuted(localTrack.jitsiTrack, muted)
             .catch(error => {
-                console.error(`setTrackMuted(${action.muted}) failed`, error);
-
-                const setMuted
-                    = mediaType === MEDIA_TYPE.AUDIO
-                        ? setAudioMuted : setVideoMuted;
+                console.error(`setTrackMuted(${String(muted)}) failed`, error);
 
                 // Failed to modify muted state - dispatch rollback action
-                store.dispatch(setMuted(!action.muted));
+                if (mediaType === MEDIA_TYPE.AUDIO) {
+                    store.dispatch(setAudioMuted(!muted));
+                } else {
+                    store.dispatch(setVideoMuted(!muted, action.muteType));
```

Comment:
Sorry about that, this was a leftover from the first approach I took. I don't understand the carbon copy part though, I adapted existing code.